sos/plugins/gluster.py

```diff
@@ -91,38 +84,40 @@ def setup(self):
             "/var/run/gluster/shared_storage/nfs-ganesha/"
         ] + glob.glob('/var/run/gluster/*tier-dht/*'))
 
-        # collect logs - apply log_size for any individual file
-        # all_logs takes precedence over logsize
         if not self.get_option("all_logs"):
-            limit = self.get_option("log_size")
-        else:
-            limit = 0
-
-        if limit:
-            for f in (glob.glob("/var/log/glusterfs/*log") +
-                      glob.glob("/var/log/glusterfs/*/*log") +
-                      glob.glob("/var/log/glusterfs/geo-replication/*/*log")):
-                self.add_copy_spec(f, limit)
+            self.add_copy_spec([
+                "/var/log/glusterfs/*log",
+                "/var/log/glusterfs/*/*log",
+                "/var/log/glusterfs/geo-replication/*/*log"
+            ])
         else:
             self.add_copy_spec("/var/log/glusterfs")
 
         if self.get_option("dump"):
-            self.make_preparations(self.statedump_dir)
-            if self.check_ext_prog("killall -USR1 glusterfs glusterfsd"):
-                # let all the processes catch the signal and create
-                # statedump file entries.
-                time.sleep(1)
-                self.wait_for_statedump(self.statedump_dir)
-                self.add_copy_spec('/tmp/glusterdump.options')
-                self.add_copy_spec(self.statedump_dir)
+            if os.path.exists(self.statedump_dir):
+                if self.check_ext_prog(
+                        "killall -USR1 glusterfs glusterfsd glusterd"):
+                    # let all the processes catch the signal and create
+                    # statedump file entries.
+                    time.sleep(1)
+                    self.wait_for_statedump(self.statedump_dir)
+                    self.add_copy_spec(self.statedump_dir)
```

Comment:
<gluster_noob_question>
How big the statedumps could be? Shall not we concern about sizelimit?
</gluster_noob_question>

Reply:
On a test environment here that supports a RHHI-V (RHV + gluster) deployment, the statedumps are all pretty small - under 100K. I don't think we need to worry about limiting here, at least not yet anyways.